Cargo.toml

```diff
@@ -0,0 +1,25 @@
+[package]
+name = "nats-source"
+version = "0.0.0"
+edition = "2021"
+publish = false
+license = "Apache 2.0"
+description = "Fluvio Nats source connector"
+
+[dependencies]
+async-trait = { version = "0.1", default-features = false}
+futures = { version = "0.3", default-features = false}
+anyhow = { version = "1.0" }
+async-std = { version = "1.12",  default-features = false, features = ["attributes", "tokio1"]}
+serde = {version = "1.0", default-features = false, features = ["derive"]}
+serde_json = { version = "1.0", default-features = false, features = ["preserve_order"] }
+thiserror = { version = "1.0", default-features = false }
+
+fluvio = { git = "https://github.com/infinyon/fluvio", rev = "d9a76327b674614c7373a9c12eb6a0ed2fcc4dc0" }
```

Comment:
Unless there is a reason, we should use fluvio deps from crates.io

Reply:
This is how it was in other connectors currently so I just used that. 
Worth mentioning this as well:
`    Updating git repository https://github.com/galibey/openapiv3`
Can switch to crates but lmk

Reply:
>  This is how it was in other connectors currently so I just used that.

There was a need to pull some latest changes from `fluvio` dep while it had not been released yet back then. But now, there is no such need, hence we should use crates.io

> Worth mentioning this as well:
>  Updating git repository https://github.com/galibey/openapiv3
> Can switch to crates but lmk consistency would be good

How is that related to the topic? That fork has customized changes that are not present in the published to crates.io version.

Reply:
I think this stems from a bug in our current `cdk generate` code project template.  I filed an issue here that discusses it more https://github.com/infinyon/fluvio/issues/3785

My recommendation is to make the fluvio dependencies a reference to the crate version e.g. `fluvio = "0.11"`

Reply:
Ok might want to update these too that point to git as well that I followed:
- [mqtt-connector](https://github.com/infinyon/mqtt-connector/blob/main/Cargo.toml#L24) - git commit ref
- [sql-connector](https://github.com/infinyon/sql-connector/blob/main/Cargo.toml#L15) - git commit ref
- [http](https://github.com/infinyon/http-source-connector/blob/main/Cargo.toml#L16) - uses tag instead commit
- [kafka](https://github.com/infinyon/kafka-connector/blob/main/Cargo.toml#L20) - uses git commit ref
It's not just cdk generate it's in every connector like that :]

Reply:
btw There is no fluvio-connector-common in crates.io.. ? I remember the decision was not to publish these

Reply:
Also cdk generate today uses git ref:
```
$ cdk generate foobar
Generating new Connector project: foobar
🤷   Please set a group name: yak
✔ 🤷   Which type of Connector would you like [source/sink]? · source
✔ 🤷   Will your Connector be public? · true
[1/8]   Done: .gitignore                                                                                                                                                                                               [2/8]   Done: Cargo.toml                                                                                                                                                                                               [3/8]   Done: Connector.toml                                                                                                                                                                                           [4/8]   Done: README.md                                                                                                                                                                                                [5/8]   Done: sample-config.yaml                                                                                                                                                                                       [6/8]   Done: src/config.rs                                                                                                                                                                                            [7/8]   Done: src/main.rs                                                                                                                                                                                              [8/8]   Done: src                                                                                                                                                                                                      [foobar@localhost infinyon]$ grep fluvio foobar/Cargo.toml 
fluvio = { git = "https://github.com/infinyon/fluvio", rev = "f9238646dd1b5197977b9c03feba39a78ef89bb8" }
fluvio-connector-common = { git = "https://github.com/infinyon/fluvio", rev = "f9238646dd1b5197977b9c03feba39a78ef89bb8", features = ["derive"]}
```

Reply:
I raised issue
- #4 
It refers to fluvio:
- https://github.com/infinyon/fluvio/issues/3785

Reply:
Since cannot address this in this PR will mark as resolved given it's consistent with everything else

Reply:
Don't worry about other places. When we touch them, we will fix them. If there is somewhere tech debt, it does not mean we should introduce another one, right? 
For `fluvio-connector-common` since it is not published, we can use the released version `0.11.2` from tag instead of a particular commit.

Reply:
Due to the types incompatibility issue, looks like it is easier for both of them to point to the git repo. 
Let's use the release version for that:
```
fluvio = { git = "https://github.com/infinyon/fluvio", tag = "v0.11.2" }
fluvio-connector-common = { git = "https://github.com/infinyon/fluvio", tag = "v0.11.2", features = ["derive"] }
```